src/plugins/intel_gpu/src/plugin/ops/condition.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "intel_gpu/plugin/program.hpp"
+#include "ngraph/op/if.hpp"
+#include "ie_ngraph_utils.hpp"
+
+#include "intel_gpu/primitives/condition.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+const size_t idx_true = 0;
+const size_t idx_false = 1;
+
+static cldnn::condition::branch gen_branch(Program& p, const std::shared_ptr<ngraph::op::v8::If>& op, size_t idx) {
+    cldnn::condition::branch branch;
+    const auto& internal_body = (idx == idx_true)? op->get_then_body() : op->get_else_body();
+
+    InferenceEngine::CNNNetwork body_network(internal_body);
+    {
+        // CNNNetwork change the input/output data type to fp32 when input/output data type is fp16
+        // To run internal body, rollback input/output data to original one.
+        size_t tidx = 0;
+        auto& model_inputs = internal_body->get_parameters();
+        for (auto& in : body_network.getInputsInfo()) {
+            auto input_data_type = InferenceEngine::details::convertPrecision(model_inputs[tidx++]->get_output_tensor(0).get_element_type());
+            if (in.second->getPrecision() != input_data_type)
+                in.second->setPrecision(input_data_type);
+        }
+
+        tidx = 0;
+        for (auto& out : body_network.getOutputsInfo()) {
+            const auto& model_output = internal_body->get_output_op(tidx++);
+            auto output_data_type = InferenceEngine::details::convertPrecision(model_output->get_output_tensor(0).get_element_type());
+            if (out.second->getPrecision() != output_data_type)
+                out.second->setPrecision(output_data_type);
+        }
+    }
+
+    auto config = p.get_config();
+    config.set_property(ov::intel_gpu::max_dynamic_batch(1));
```

Comment:
Don't we need to change `compilation_num_threads` too? Also, I think we should consider sharing task executor of inner and outer network (may be done separately)

Reply:
the case to share task executor and impl_cache will be updated at next step.

Reply:
I think the config from Program::get_config() already has the property for **compilation_num_threads**. 
So I just to add the property for max_dynamic_batch to disable dynamic batch for inner program build.
Is it necessary additional setting for **compilation_num_threads** ?

Reply:
> I think the config from Program::get_config() already has the property for compilation_num_threads.

It has, but I assume that the value is same as for outer network which means that each inner program will create task executor with `compilation_num_threads` threads, so we'll have `(num_inner_networks + 1) * compilation_num_threads` threads created which is probably not what we want.

Reply:
Fixed codes to share single task_executor between all programs including inner programs. thanks.